src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -11699,6 +11742,1122 @@ class Base
+}
+
+unsafe class C1(int x, S s) : Base(() => 
+                                   {
+                                       _ = x + s.f;
+                                   })
+{
+    int* p1 = &x;
+    int* p2 = &s.f;
+
+    void M1()
+    {
+        _ = x + s.f;
+    }
+}
+
+struct S
+{
+    public int f;    
+}
+";
+            CreateCompilation(text, options: TestOptions.UnsafeReleaseDll).VerifyEmitDiagnostics(
+                // (12,15): error CS0212: You can only take the address of an unfixed expression inside of a fixed statement initializer
+                //     int* p1 = &x;
+                Diagnostic(ErrorCode.ERR_FixedNeeded, "&x").WithLocation(12, 15),
+                // (13,15): error CS0212: You can only take the address of an unfixed expression inside of a fixed statement initializer
+                //     int* p2 = &s.f;
+                Diagnostic(ErrorCode.ERR_FixedNeeded, "&s.f").WithLocation(13, 15)
+                );
+        }
+
+        [Fact]
+        public void ParameterCapturing_093_AddressOfCapturedParameters()
+        {
+            var text = @"
+unsafe class Base
+{
+    public Base(int* x, int* y) {}
+}
+
+unsafe class C1(int x, S s) : Base(&x, &s.f)
+{
+    int y = M(() => 
+              {
+                  _ = x + s.f;
+              });
+
+    static int M(System.Action a)
+    {
+        return 0;
+    }
+
+    void M1()
+    {
+        _ = x + s.f;
+    }
+}
+
+struct S
+{
+    public int f;    
+}
+";
+            CreateCompilation(text, options: TestOptions.UnsafeReleaseDll).VerifyEmitDiagnostics(
+                // (7,36): error CS0212: You can only take the address of an unfixed expression inside of a fixed statement initializer
+                // unsafe class C1(int x, S s) : Base(&x, &s.f)
+                Diagnostic(ErrorCode.ERR_FixedNeeded, "&x").WithLocation(7, 36),
+                // (7,40): error CS0212: You can only take the address of an unfixed expression inside of a fixed statement initializer
+                // unsafe class C1(int x, S s) : Base(&x, &s.f)
+                Diagnostic(ErrorCode.ERR_FixedNeeded, "&s.f").WithLocation(7, 40)
+                );
+        }
+
+        [Fact]
+        public void ParameterCapturing_094_DefiniteAssignment()
+        {
+            var text = @"
+class C1(int x, S s, string y)
+{
+    void M1()
+    {
+        _ = x + s.f + y.Length;
+    }
+
+    void M2()
+    {
+        _ = s.f + y.Length + x;
+    }
+
+    void M3()
+    {
+        _ = y.Length + x + s.f;
+    }
+
+    void M4()
+    {
+        _ = x;
+    }
+
+    void M5()
+    {
+        _ = s.f;
+    }
+
+    void M6()
+    {
+        _ = y.Length;
+    }
+}
+
+struct S
+{
+    public int f;    
+    public S(int x) => f = x; 
+}
+";
+            CreateCompilation(text).VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void ParameterCapturing_095_DefiniteAssignment()
+        {
+            var text = @"
+class C1(out int x, out S s, out string y)
```

Comment:
Consider also testing `ref` and `in` primary constructor parameters.

Reply:
> Consider also testing ```ref``` and ```in``` primary constructor parameters.

They cannot be captured

Reply:
Will add suggested tests as ParameterCapturing_100_DefiniteAssignment and ParameterCapturing_101_DefiniteAssignment in the next PR